### PR TITLE
Controller specs

### PR DESCRIPTION
**Pull request by Busiwa24**
**In the following pull request I have completed these tasks:**
- Added RSpec as a gem to my project in order to write tests for controllers according to the task guidelines provided
- Created a Request spec file for all my controllers.
- Ensure that for each action you check:
- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/controller_test/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    end
+
+    it 'Renders index template' do
+      expect(response).to render_template('index')
+    end
+
+    it 'Return text inside index' do
+      expect(response.body).to include("This is the list of posts for the user with id of #{id}")
+    end
+  end
```
Comment (Dipeshtwis) [inline spec/controller_test/posts_controller_spec.rb:18]: - [ ] I would recommend adding the test for the `Post#show` as well.

File: spec/controller_test/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    end
+
+    it 'Renders index template' do
+      expect(response).to render_template('index')
+    end
+
+    it 'Return text inside index' do
+      expect(response.body).to include('This is the main section for the users')
+    end
+  end
```
Comment (Dipeshtwis) [inline spec/controller_test/users_controller_spec.rb:17]: - [ ] I would recommend adding the test for the `Users#show` as well.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Busiwa24, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️
 
**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controller_test/posts_controller_spec.rb (hunk 1 above)
Comment (Busiwa24) [reply spec/controller_test/posts_controller_spec.rb:18]: resolved

File: spec/controller_test/users_controller_spec.rb (hunk 2 above)
Comment (Busiwa24) [reply spec/controller_test/users_controller_spec.rb:17]: fixed

Comment (oloomoses) [reply spec/controller_test/users_controller_spec.rb:17]: - [ ] As suggested in the previous review by @Dipeshtwis, Kindly add spec test for the `user#show` the same way you did with `index`. This means that you will add another `describe` block for `#show`.

File: spec/controller_test/posts_controller_spec.rb (hunk 1 above)
Comment (oloomoses) [reply spec/controller_test/posts_controller_spec.rb:18]: - [ ] As suggested in the previous review by @Dipeshtwis, Kindly add spec test for the `post#show` the same way you did with `index`. This means that you will add another `describe` block for `#post`.

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi @Busiwa24, 

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
# Status: Approved🎉🎉

## Hi @Busiwa24, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Keep Rocking!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**